### PR TITLE
Pass `num_groups` arg to ModelValidator.fix

### PR DESCRIPTION
**Pull request by ffuuugor**
## Problem
As highlighted by #567, end user have little control over how exactly `ModelValidator.fix()` deals with BatchNorms.
For example, our approach to choosing number of groups is `gcd(32, module.num_features)`, which is fine for most cases, but can break a model occasionally (see #567 for a demonstration)

## Solution
Pass `num_groups` as kwarg, allowing clients to control the behaviour

### CODE REVIEW

File: opacus/validators/batch_norm.py [hunk 1]
```diff
@@ -86,9 +93,10 @@ def _batchnorm_to_groupnorm(module: BATCHNORM) -> nn.GroupNorm:
         A default value of 32 is chosen for the number of groups based on the
         paper *Group Normalization* https://arxiv.org/abs/1803.08494
     """
-    return nn.GroupNorm(
-        math.gcd(32, module.num_features), module.num_features, affine=module.affine
-    )
+    if num_groups is None:
+        num_groups = math.gcd(32, module.num_features)
+
+    return nn.GroupNorm(num_groups, module.num_features, affine=module.affine)
```
Comment (karthikprasad) [inline opacus/validators/batch_norm.py:99]: Might want to check that `num_features` is divisible by `num_groups`; and raise an exception otherwise.

File: opacus/validators/batch_norm.py [hunk 2]
```diff
@@ -62,16 +62,23 @@ def fix(module: BATCHNORM, **kwargs) -> Union[nn.GroupNorm, INSTANCENORM]:
...
         _batchnorm_to_instancenorm(module)
         if is_replace_bn_with_in
-        else _batchnorm_to_groupnorm(module)
+        else _batchnorm_to_groupnorm(module, num_groups)
     )
 
 
-def _batchnorm_to_groupnorm(module: BATCHNORM) -> nn.GroupNorm:
+def _batchnorm_to_groupnorm(
+    module: BATCHNORM, num_groups: Optional[int] = None
```
Comment (karthikprasad) [inline opacus/validators/batch_norm.py:80]:
```suggestion
    module: BATCHNORM, *, num_groups: Optional[int] = None
```

File: opacus/validators/batch_norm.py (hunk 1 above)
Comment (ffuuugor) [reply opacus/validators/batch_norm.py:99]: Actually I didn't do any validation on purpose - exception will be thrown from `nn.GroupNorm` constructor immediately and it will be a good one, so no reason to wrap it with our own

Comment (karthikprasad) [reply opacus/validators/batch_norm.py:99]: Point taken!

**Review by karthikprasad — APPROVED** [review]
